app/controllers/api/base_controller/providers.rb

```diff
@@ -46,6 +46,26 @@ def delete_resource_providers(type, id = nil, _data = nil)
         end
       end
 
+      def meta_data_resource_providers(_type, _id = nil, data)
+        provider = ExtManagementSystem.find_by_name(data.delete("name"))
+        data.each do |key, value|
+          attribute = provider.custom_attributes.find_by_name(key)
```

Comment:
This is an N+1.  I think you can query all of the requisite attributes up front (or just query them all)